spec/factories/omniauth.rb

```diff
@@ -0,0 +1,37 @@
+FactoryGirl.define do
+
+  factory :auth_hash, class: OmniAuth::AuthHash do
+    skip_create
+
+    transient do
+      sequence(:uid)
+      provider "provider"
+      email { "email" }
+    end
```

Comment:
As far as I understand, transient attributes are intended for situations where the factory needs to receive options that will be used during the object creation process but won't actually be properties of the created object. In this case, since you're just trying to set up some defaults, you can simply list them as normal attributes of the base `auth_hash` factory.